adaptdl/adaptdl/torch/data.py

```diff
@@ -239,10 +245,40 @@ def _sync_local_bsz(self):
         else:
             # Autoscale batch size, compute on rank 0 and broadcast.
             speedup_fn = get_speedup_fn()
-            _, local_bsz = speedup_fn(adaptdl.env.num_nodes(),
-                                      adaptdl.env.num_replicas(),
-                                      return_local_bsz=True)
-            self.current_local_bsz = adaptdl.collective.broadcast(local_bsz)
+
+            # check whether predict param is up
+            is_init = speedup_fn._params is None
+            # if first time called, initiliza the batch size field
+            if is_init:
```

Comment:
I think you can determine this case simply with `self.current_local_bsz is None`? Accessing private field `_params` is not a good practice.